edxval/serializers.py

```diff
@@ -205,3 +211,40 @@ def update(self, instance, validated_data):
             course_video.save()
 
         return instance
+
+class VideosPagination(pagination.BasePagination):
+    """
+    Paginator for APIs in edx-val.
+    """
+    invalid_page_message = 'Invalid page.'
+
+    def paginate_queryset(self, queryset, page_no=1, page_size=20):
+        """
+        Paginate the queryset and returns the interable data
+        for the given page and page_size
+        """
+        paginator = Paginator(queryset, page_size)
+
+        try:
+            self.page = paginator.page(page_no)
+        except InvalidPage as exc:
+            msg = self.invalid_page_message.format(
+                page_number=page_no, message=six.text_type(exc)
+            )
+            raise NotFound(msg)
+
+        return list(self.page)
+
+    def get_paginated_response(self, data, sort_dir, sort_field):
+        """
+        Annotate the response with pagination information.
+        """
+        return {
+            'page_size': self.page.paginator.per_page,
+            'total_count': self.page.paginator.count,
+            'total_pages': self.page.paginator.num_pages,
+            'current_page': self.page.number,
+            'sort_field': sort_field.value,
+            'sort_dir': sort_dir.value,
+            'results': data
```

Comment:
this is using only the fields required by pagination header/footer on the front end side. for example next and previous page links would be redundant if added.